server.js

```diff
@@ -1,35 +1,308 @@
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
-mongoose.Promise = Promise;
+import express from "express"
+import cors from "cors"
+import mongoose from "mongoose"
+
+import topMusicData from "./data/top-music.json"
+
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo-tiiliu"
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.Promise = Promise
+
 
 // Defines the port the app will run on. Defaults to 8080, but can be overridden
 // when starting the server. Example command to overwrite PORT env variable value:
 // PORT=9000 npm start
-const port = process.env.PORT || 8080;
-const app = express();
+const port = process.env.PORT || 8080
+const app = express()
+
+
+const Track = mongoose.model("Track", {
+  id: Number,
+  trackName: String,
+  artistName: String,
+  genre: String,
+  bpm: Number,
+  energy: Number,
+  danceability: Number,
+  loudness: Number,
+  liveness: Number,
+  valence: Number,
+  length: Number,
+  acousticness: Number,
+  speechiness: Number,
+  popularity: Number
+})
+
+if (process.env.RESET_DB) {
+
+  // Only for setting things up locally
+  const seedDatabase = async () => {
+    await Track.deleteMany({})
+
+    topMusicData.forEach((track) => {
+      const newTrack = new Track(track)
+      newTrack.save()
+    })
+  }
+  seedDatabase()
+}
+
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors());
-app.use(express.json());
+app.use(cors())
+app.use(express.json())
+
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({
+      error: "Service unavailable",
+    })
+  }
+})
+
 
-// Start defining your routes here
+// Description for the routes
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
-});
+  res.send({
+    "Top Music API":"Serving you info about Spotifys top tracks",
+    "Routes":[{
+      "/tracks":"Get all tracks",
+      "/tracks/high-tempo":"Get all tracks with high tempo",
+      "/tracks/danceable":"Get all tracks which are danceable",
+      "/tracks/random-track":"Get a random track",
+      "/tracks/genres/:genre":"Get tracks by genre (for example: /tracks/genres/latin)",
+      "/tracks/artists/:artist":"Get a track by the artists name (for example: /tracks/artists/ed sheeran)",
+      "/tracks/songtitles/:songtitle":"Get a specific track by the songtitle (for example: /tracks/songtitles/truth hurts)",
+      "/tracks/id/:id":"Get a track by _id (for example: /tracks/id/6283e58592aafedccbeb0e0e)",
+      "/tracks?genre=${genre}":"Get tracks by genre (for example: /tracks?genre=hip hop)",
+      "/tracks?artist=${artist}":"Get all tracks from a specific artist (for example: /tracks?artist=billie eilish)",
+      "/tracks?songtitle=${songtitle}":"Get a specific track by the songtitle (for example: /tracks?songtitle=shallow)"
+    }]
+  })
+})
```

Comment:
You are showing all your endpoints very clearly and with descriptions, that is very nice!
It can be a lot to write, you could try out using the npm package express list endpoints. It shows all the endpoints that you create automatically. Then you only need to document in the readme what endpoints you have!